subprojects/platform-jvm/src/main/java/org/gradle/jvm/toolchain/internal/JavaToolchainComparator.java

```diff
@@ -23,10 +23,17 @@
 
 public class JavaToolchainComparator implements Comparator<JavaToolchain> {
 
+    private final boolean currentJvmMatters;
+
+    public JavaToolchainComparator(boolean currentJvmMatters) {
+        this.currentJvmMatters = currentJvmMatters;
```

Comment:
This is not needed. The current vm detector is disabled if auto-detection is disabled.

Reply:
I don't think this is true. The `JavaToolchainQueryService` will sort the installations via the `JavaToolchainComparator` even when auto-detection is off. Let's say on a CI server where the installations have been specified via `org.gradle.java.installations.paths`. If I understand correctly, in such a situation we don't want the current JVM to end up first/be picked, but if I remove that boolean flag, it will be. 

As proof: `JavaToolchainQueryServiceTest.refer version Gradle is running on as long as it is a match` passes even if you set `auto-detection` to `false` on line 393. (Talking about the code as is now in this PR).

Reply:
You are right! I missed that corner case. This should be clearly documented though. I will add a note to the issue about this.